aws/resource_aws_worklink_fleet.go

```diff
@@ -0,0 +1,399 @@
+						},
+					},
+				},
+			},
+			"device_ca_certificate": {
+				Type:     schema.TypeString,
+				Optional: true,
+				StateFunc: func(v interface{}) string {
+					s, ok := v.(string)
+					if !ok {
+						return ""
+					}
+					return strings.TrimSpace(s)
+				},
+			},
+			"identity_provider": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"saml_metadata": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringLenBetween(1, 204800),
+						},
+					},
+				},
+			},
+			"company_code": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_updated_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"optimize_for_end_user_location": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+		},
+	}
+}
+
+func resourceAwsWorkLinkFleetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).worklinkconn
+
+	input := &worklink.CreateFleetInput{
+		FleetName:                  aws.String(d.Get("name").(string)),
+		OptimizeForEndUserLocation: aws.Bool(d.Get("optimize_for_end_user_location").(bool)),
+	}
+
+	if v, ok := d.GetOk("display_name"); ok {
+		input.DisplayName = aws.String(v.(string))
+	}
+
+	resp, err := conn.CreateFleet(input)
+	if err != nil {
+		return fmt.Errorf("Error creating Worklink Fleet: %s", err)
+	}
+
+	d.SetId(aws.StringValue(resp.FleetArn))
+	return resourceAwsWorkLinkFleetUpdate(d, meta)
+}
+
+func resourceAwsWorkLinkFleetRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).worklinkconn
+
+	resp, err := conn.DescribeFleetMetadata(&worklink.DescribeFleetMetadataInput{
+		FleetArn: aws.String(d.Id()),
+	})
+	if err != nil {
+		if isAWSErr(err, worklink.ErrCodeResourceNotFoundException, "") {
+			log.Printf("[WARN] Worklink Fleet (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error describe worklink fleet: %s", err)
+	}
+
+	d.Set("arn", d.Id())
+	d.Set("name", aws.StringValue(resp.FleetName))
+	d.Set("display_name", aws.StringValue(resp.DisplayName))
+	d.Set("optimize_for_end_user_location", aws.BoolValue(resp.OptimizeForEndUserLocation))
+	d.Set("company_code", aws.StringValue(resp.CompanyCode))
+	d.Set("created_time", resp.CreatedTime.Format(time.RFC3339))
+	if resp.LastUpdatedTime != nil {
+		d.Set("last_updated_time", resp.LastUpdatedTime.Format(time.RFC3339))
+	}
+	auditStreamConfigurationResp, err := conn.DescribeAuditStreamConfiguration(&worklink.DescribeAuditStreamConfigurationInput{
+		FleetArn: aws.String(d.Id()),
+	})
+	if err != nil {
+		if !isAWSErr(err, worklink.ErrCodeResourceNotFoundException, "") {
```

Comment:
If the error is `ErrCodeResourceNotFoundException` what value does `auditStreamConfigurationResp` have? The below code is assuming it is not `nil` by dereferencing it, which will cause a panic if it is `nil`.

Reply:
Fix to just return err.